### ISSUE TITLE
Slow start to net log parsing in 1.10

### ISSUE THREAD

**Issue by travellingkiwi** [EDDiscovery/EDDiscovery-Mac#7]
Netlog parsing is still very slow when a large number of systems are known. On a MacPro 2013 with 32GB of memory it can take several minutes to start

I've traced it to the

`- (void)scanNetLogFilesDir`

routine. In there it does

```
      if (netLogFile.complete == NO) {
        if ([netLog isEqualToString:netLogsFilesAttrs.lastObject[FILE_KEY]] == NO && allSystems == nil) {
          allSystems = [[System allSystemsInContext:WORK_CONTEXT] mutableCopy];
          allNames   = [NSMutableArray arrayWithCapacity:allSystems.count];
          allJumps   = [[Jump allJumpsOfCommander:commander] mutableCopy];
          for (System *system in allSystems) {
            [allNames addObject:system.name];
          }
        }

        netLogFile.complete = YES;

        numJumps += [self parseNetLogFile:netLogFile systems:allSystems names:allNames jumps:allJumps lastJump:&lastJump];

        numParsed++;

        [MAIN_CONTEXT performBlock:^{
          LoadingViewController.progressIndicator.doubleValue = progressValue;
        }];
      }

```

So there's the call to get all the systems into an array, then all names... then all your jumps (That one at least is reasonably quick) and then for EVERY system known it goes through and adds the names to the allNames array... 

When there's 2.5million systems this takes a while... e.g.

`2016-09-15 07:58:13.291 EDDiscovery[68095:3799307] Parsed 1 jumps from 2 netLog files in 213.9 seconds
`
213.9 seconds (Almost 4 minutes) to parse 1 jump from 2 net log files... 

I put some logging in there to show how long it can take... (This is with 5 Million systems in the database BTW) e.g.

```
2016-09-15 07:54:39.512 EDDiscovery[68095:3800519] -[NetLogParser scanNetLogFilesDir]: Parsing netLog file: netLog.160913181331.01.log
2016-09-15 07:54:39.512 EDDiscovery[68095:3800519] -[NetLogParser scanNetLogFilesDir]: Grabbing allSystemsInContext
2016-09-15 07:55:04.879 EDDiscovery[68095:3800519] -[NetLogParser scanNetLogFilesDir]: gettingAllNames
2016-09-15 07:55:04.888 EDDiscovery[68095:3800519] -[NetLogParser scanNetLogFilesDir]: gettingAllJumps
2016-09-15 07:55:04.892 EDDiscovery[68095:3800519] -[NetLogParser scanNetLogFilesDir]: Performing strange loop for adding systems to allNames
2016-09-15 07:58:13.040 EDDiscovery[68095:3800519] -[NetLogParser scanNetLogFilesDir]: Strange Loop Completed
```

It doesn't run EVERY startup. However I'm struggling to understand WHY it's doing this. And in the MainContext too (It wouldn't be so bad if it were backgrounded. But it's not).

**Comment by noberasco** [comment]
Prior to ED patch 2.1, which added system coords to netlog files, there was a limited amount of systems with known coordinates on EDSM. After patch 2.1 this number exploded exponentially.

The netlog parsing code is (roughly) as follows:
1 - for each netlog file that still needs parsing
2 - search for 'jump to system' logs entries
3 - add jump records as necessary
4 - add or update system records as necessary

In case multiple netlog files need to be parsed (such as initial parsing of ED log dir), the code you highlighted above speeded parsing up by pre-fetching all local systems into RAM. This is evidently no longer the case: after the explosion in the number of systems the time needed to load all systems in memory is now way higher than the subsequent parsing speed improvement.

Luckily, this pre-fetching of systems is not necessary to successfully parse the netlog files, I'll try to comment it out.

**Comment by noberasco** [comment]
@travellingkiwi, please give this commit a test run as well :-)

**Comment by noberasco** [comment]
This fix needs some more working on. Stay tuned...

**Comment by noberasco** [comment]
Should be good to go now...

**Comment by noberasco** [comment]
Hopefully resolved with 426b0e615d9e3c373d00923e92d0e24c475f5cac
